packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -808,8 +811,16 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         return this.reSubmit;
     }
 
+    public get disposeFn(): (error?: ICriticalContainerError) => void {
+        return this.context.disposeFn ?? this.context.closeFn;
+    }
+
     public get closeFn(): (error?: ICriticalContainerError) => void {
-        return this.context.closeFn;
+        // Also call disposeFn to retain functionality of runtime being disposed on close
```

Comment:
This is a somewhat temporary measure, right?  Like in next major version (per n/n-1 support) this can be changed to only close?  Is that follow-up tracked?

Reply:
I think there are some GC features that depend on the disposition on close, so those would also need to be adjusted. [AB#2953](https://dev.azure.com/fluidframework/internal/_workitems/edit/2953)